fs2/prediction_writing_callback.py

```diff
@@ -143,22 +50,43 @@ def get_synthesis_output_callbacks(
             raise ValueError(
                 "We cannot synthesize waveforms without a vocoder. Please ensure that a vocoder is specified."
             )
-        callbacks.append(
-            PredictionWritingWavCallback(
+        callbacks[SynthesizeOutputFormats.wav] = PredictionWritingWavCallback(
+            config=config,
+            device=device,
+            global_step=global_step,
+            output_dir=output_dir,
+            output_key=output_key,
+            vocoder_model=vocoder_model,
+            vocoder_config=vocoder_config,
+            vocoder_global_step=vocoder_global_step,
+        )
+    if SynthesizeOutputFormats.spec in output_type:
+        callbacks[SynthesizeOutputFormats.spec] = PredictionWritingSpecCallback(
+            config=config,
+            global_step=global_step,
+            output_dir=output_dir,
+            output_key=output_key,
+        )
+    if SynthesizeOutputFormats.textgrid in output_type:
```

Comment:
ah yes, this is much better and makes us more extendable in the future. nicely done @joanise !